### DISCUSSION TITLE
Meltano

### DISCUSSION THREAD

**Discussion by marijaselakovic** [crate/crate-clients-tools#196] [Ideas]
One of the significant issues when working with CrateDB is data migration from other data sources. There is lack of documentation and tutorials on how data can be imported to CrateDB from other open-source or commercial systems.

There has been already a great effort on providing CrateDB support for meltano project: https://github.com/crate-workbench/meltano-target-cratedb Once we have a clear idea which migrations we can illustrate, we should write 1 or more tutorials and publish it on community.

**Comment by amotl** [comment]
Let's collect a few spots where communities and organizations huddle taps and targets.

- https://github.com/Widen/dagster-meltano-example (https://engineering.widen.com/blog/Dagster-+-Meltano/)
- https://github.com/Widen/tap-rest-api-msdk
- https://github.com/dbt-labs/tap-s3-csv
- https://github.com/reservoir-data
- https://github.com/celine-eu
- https://github.com/Matatika

**Comment by amotl** [comment]
The canonical pipeline source component (tap) to read CSV and Excel files is [tap-spreadsheets-anywhere](https://github.com/ets/tap-spreadsheets-anywhere). On the other hand, we just discovered [tap-spreadsheets](https://github.com/celine-eu/tap-spreadsheets), which already supports [Meltano SDK's Inline Stream Maps](https://sdk.meltano.com/en/latest/stream_maps.html). Thanks, @lc-spxl!

Contrary to the former which uses [smart-open](https://pypi.org/project/smart-open/) for universal file access, the latter uses the fsspec-based [s3fs](https://pypi.org/project/s3fs/) package. If fsspec is fancied over smart-open, [universal-pathlib](https://universal-pathlib.readthedocs.io/) could come to the rescue to provide the same kind of polyglottery.

/cc @edgarrmondragon

**Reply by lc-spxl** [reply]
Hi @amotl @marijaselakovic, Luca from the CELINE project here, thank you for the mention. 

Given that we are in early stage, we also have a pipeline setup based on meltano, dbt and orchestrated with prefect-based python flows. It can run via a CLI command, for one shot data sync.

I leave here the docs for your convenience https://celine-eu.github.io/projects/celine-utils/docs/pipeline-tutorial/

**Reply by amotl** [reply]
Hi Luca, nice to meet you. Thanks for your swift reply.

The CELINE project, the selected [ontologies](https://celine-eu.github.io/ontologies/) and larger paradigms look very promising to us. We are looking forward to better join the Meltano ecosystem with [CrateDB 6.2](https://github.com/crate-workbench/meltano-target-cratedb/pull/57), aiming for a release in February if everything goes well. Before that, due to a silly workaround, the [meltano-target-cratedb](https://github.com/crate-workbench/meltano-target-cratedb) package wasn't really interoperable.

Kudos for implementing a [medallion-style](https://dataengineering.wiki/Concepts/Data+Architecture/Medallion+Architecture) workflow around [those](https://celine-eu.github.io/projects/celine-utils/docs/pipeline-tutorial/#kubernetes-production-deployment) excellent open source components. This certainly sparks our interest, as we only integrated particular applications with CrateDB yet, like [dbt](https://cratedb.com/docs/guide/integrate/dbt/), [Prefect](https://cratedb.com/docs/guide/integrate/prefect/), and [Marquez](https://cratedb.com/docs/guide/integrate/marquez/), without connecting them together well.

We will be happy to exercise your pipeline setup with CrateDB instead of PostgreSQL and report back how that works, if you like this idea. I am sure we might face the one or the other obstacle along the way, but CrateDB 6.2 will also be [nicer to dbt](https://github.com/crate/dbt-cratedb2/pull/32) already. 🍀
